components/widgets/menu/menu_items/menu_item.tsx

```diff
@@ -35,9 +35,11 @@ export default function menuItem(Component: React.ComponentType<any>) {
                 );
             }
 
+            const iconClassname = icon ? ' MenuItem--with-icon' : '';
+
             return (
                 <li
-                    className='MenuItem'
+                    className={`MenuItem${iconClassname}`}
```

Comment:
Another way you could do this is using the `classnames` utility that we've started using elsewhere. It's a bit easier than fiddling around with appending strings and making sure spaces exist properly
```
className={classnames('MenuItem', {
    'MenuItem--with-icon': icon,
})
```